packages/framework/src/Model/Order/Item/OrderItemDataFactory.php

```diff
@@ -52,4 +73,30 @@ protected function addFieldsByOrderItemType(OrderItemData $orderItemData, OrderI
             $orderItemData->payment = $orderItem->getPayment();
         }
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData $orderItemData
+     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItem $orderItem
+     * @return bool
+     */
+    protected function canUsePriceCalculation(OrderItemData $orderItemData, OrderItem $orderItem): bool
+    {
+        if ($this->orderItemPriceCalculation === null) {
+            return false;
+        }
+
+        $calculatedPriceWithoutVat = $this->orderItemPriceCalculation->calculatePriceWithoutVat($orderItemData);
```

Comment:
I don't understand why is the function so complicated. Shouldn't it just return `return !$orderItem->hasForcedTotalPrice()`?

Reply:
I implemented the `OrderItemDataFactory::canUsePriceCalculation()` method so that it uses price calculation always if possible so that the administration is easier. If the total prices is "forced" to be the same value that would be actually calculated by price calculation, it will still suggest to use the price calculation.

Only when the prices are inconsistent with the price calculation implementation it will set the `OrderItemData::$usePriceCalculation` to `false`.

Considering that I have to specify the wnated behavior to you I guess I should write a test for it, so that these expectations are explicit, right?

Reply:
Let's assume following case:
* I open an order in administration
* uncheck "use price calculation" on order Item XXX
* save
* 2 total prices will be saved together with XXX
* I reopen the same order
* the "use price calculation" is unchecked (from user-point of view WHY?!)
* save
* 2 total prices are removed from XXX

Am I correct?
I don't assume this to be either user-friendly, reasonable scenario

Reply:
Yes, this would be the behavior. I don't really know whether it is acceptable or user-friendly.

Reply:
It's doing "something" when I didn't change anything. It isn't reasonable for me. Please discuss it with our Product Owner

Reply:
After discussion with @simara-svatopluk  this behavior is looking strange to me. Please can you describe use-case that create need for such behavior?
In my point of view, after migrating to this version, I would set all existing order items to "do not calculate price" setting. But new ones will use calculation. It is clear, consistent and easily explainable.

Reply:
I've changed the behavior in 3cf8acd96